src/ophys_etl/modules/segmentation/graph_utils/creation.py

```diff
@@ -39,8 +35,14 @@ def create_graph(row_min: int, row_max: int, col_min: int, col_max: int,
         kernel = list(itertools.product([-1, 0, 1], repeat=2))
         kernel.pop(kernel.index((0, 0)))
 
-    rows, cols = np.mgrid[row_min:(row_max + 1), col_min:(col_max + 1)]
     graph = nx.Graph()
+
+    if (row_min == row_max) & (col_min == col_max):
+        # trivial case with no edges and 1 node
+        graph.add_node((row_min, col_min))
+        return graph
+
+    rows, cols = np.mgrid[row_min:(row_max + 1), col_min:(col_max + 1)]
```

Comment:
If this method does not add any edge attributes, should it even create any edges at all? Do we want to move the kernel kwarg to `add_pearson_edge_attributes` so that, if we ever want to try doing Pearson correlation beyond just the 8 nearest neighbors, we only have to change the kernel in that method.

Reply:
I was intentionally trying to separate the structural definition of the graph (this method) from the computations of graph/edge attributes.
This makes some sense to me from a modularity perspective, and let me write a more generic partition method for multiprocessing.
In your use case, we'd just create a new graph structure and pass it to the Pearson function.
To some degree, I do have these two things bundled together as a single call:
https://github.com/AllenInstitute/ophys_etl_pipelines/blob/b10adefa943999d74b395210a58415e6d4c6cab0/src/ophys_etl/modules/segmentation/modules/calculate_edges.py#L33-L37
I could make that more flexible such that `CalculateEdgesInputSchema` has an optional Nested `CreateGraphInputSchema` to give full control to the integrated call.

Reply:
I think I missed that `networkx.neighbors` follows graph edges rather than a Euclidean definition of neighbors. I agree. What you have done makes ssense.